torch/distributed/_composable/checkpoint_activation.py

```diff
@@ -26,7 +30,19 @@ def _no_hook(module: nn.Module):
 
 class _ModuleHookCheckpointFunction(torch.autograd.Function):
     @staticmethod
-    def forward(ctx, module: nn.Module, output: Any, *inputs: Any) -> Any:  # type: ignore[override]
+    def forward(
+        ctx,
+        module_fn_with_flat_args: Callable,
+        module: nn.Module,
```

Comment:
Question for myself: how come this checkpoint function needs access to the module? (will keep reading)

Reply:
It looks like, with this change, you don't technically access to the module, you just need access to checkpoint.state(module)